packages/SwingSet/test/virtualObjects/test-virtualObjectManager.js

```diff
@@ -91,88 +91,113 @@ test('virtual object operations', t => {
-    ['o+1/4', thingVal(300, 'thing-4', 0)],
+    ['o+1/1', thingVal(0, 'thing-1', 0)], // t1-0
+    ['o+1/2', thingVal(100, 'thing-2', 0)], // t2-0
+    ['o+1/3', thingVal(200, 'thing-3', 0)], // t3-0
+    ['o+1/4', thingVal(300, 'thing-4', 0)], // t4-0
   ]);
 
   // phase 2: first batch-o-stuff
-  t.is(thing1.inc(), 1);
-  t.is(zot1.sayHello('hello'), 'hello Alice');
-  t.is(thing1.inc(), 2);
-  t.is(zot2.sayHello('hi'), 'hi Bob');
-  t.is(thing1.inc(), 3);
-  t.is(zot3.sayHello('aloha'), 'aloha Carol');
-  t.is(zot4.sayHello('bonjour'), 'bonjour Dave');
-  t.is(zot1.sayHello('hello again'), 'hello again Alice');
+  t.is(thing1.inc(), 1); // [t1-1 z4-0 z3-0 z2-0] evict z1-0
+  // t1-1: 'thing-1' 1 0
+  t.is(zot1.sayHello('hello'), 'hello Alice'); // [z1-1 t1-1 z4-0 z3-0] evict z2-0
+  // z1-1: 23 'Alice' 'is this on?' 1
+  t.is(thing1.inc(), 2); // [t1-2 z1-1 z4-0 z3-0]
+  // t1-2: 'thing-1' 2 0
+  t.is(zot2.sayHello('hi'), 'hi Bob'); // [z2-1 t1-2 z1-1 z4-0] evict z3-0
+  // z2-1: 29 'Bob' 'what are you saying?' 1
+  t.is(thing1.inc(), 3); // [t1-3 z2-1 z1-1 z4-0]
+  // t1-3: 'thing-1' 3 0
+  t.is(zot3.sayHello('aloha'), 'aloha Carol'); // [z3-1 t1-3 z2-1 z1-1] evict z4-0
+  // z3-1: 47 'Carol' 'as if...' 1
+  t.is(zot4.sayHello('bonjour'), 'bonjour Dave'); // [z4-1 z3-1 t1-3 z2-1] evict z1-1
+  // z4-1: 66 'Dave' 'you and what army?' 1
+  t.is(zot1.sayHello('hello again'), 'hello again Alice'); // [z1-2 z4-1 z3-1 t1-3] evict z2-1
+  // z1-2: 23 'Alice' 'is this on?' 2
   t.is(
-    thing2.describe(),
+    thing2.describe(), // [t2-0 z1-2 z4-1 z3-1] evict t1-3
     'thing-2 counter has been reset 0 times and is now 100',
   );
   t.deepEqual(dumpStore(), [
-    ['o+1/1', thingVal(3, 'thing-1', 0)],
-    ['o+1/2', thingVal(100, 'thing-2', 0)],
-    ['o+1/3', thingVal(200, 'thing-3', 0)],
-    ['o+1/4', thingVal(300, 'thing-4', 0)],
-    ['o+2/1', zotVal(23, 'Alice', 'is this on?', 2)],
-    ['o+2/2', zotVal(29, 'Bob', 'what are you saying?', 0)],
-    ['o+2/3', zotVal(47, 'Carol', 'as if...', 0)],
-    ['o+2/4', zotVal(66, 'Dave', 'you and what army?', 0)],
+    ['o+1/1', thingVal(3, 'thing-1', 0)], // =t1-3
+    ['o+1/2', thingVal(100, 'thing-2', 0)], // =t2-0
+    ['o+1/3', thingVal(200, 'thing-3', 0)], // =t3-0
+    ['o+1/4', thingVal(300, 'thing-4', 0)], // =t4-0
+    ['o+2/1', zotVal(23, 'Alice', 'is this on?', 1)], // =z1-1
+    ['o+2/2', zotVal(29, 'Bob', 'what are you saying?', 1)], // =z2-1
+    ['o+2/3', zotVal(47, 'Carol', 'as if...', 0)], // =z3-0
+    ['o+2/4', zotVal(66, 'Dave', 'you and what army?', 0)], // =z4-0
   ]);
 
   // phase 3: second batch-o-stuff
-  t.is(thing1.get(), 3);
-  t.is(thing1.inc(), 4);
-  t.is(thing4.reset(1000), 1);
-  t.is(zot3.rename('Chester'), 'Chester');
-  t.is(zot1.getInfo(), 'zot Alice tag=is this on? count=3 arbitrary=23');
-  t.is(zot2.getInfo(), 'zot Bob tag=what are you saying? count=2 arbitrary=29');
+  t.is(thing1.get(), 3); // [t1-3 t2-0 z1-2 z4-1] evict z3-1
```

Comment:
We need these tests check that no storage operations occur when a Representative is reanimated, since that's a determinism invariants we must to maintain.

To test it, I think we need something like:
* make `const thing1` into `let thing1`
* drop the reference with `thing1 = undefined`
* do `gc()` and `await new Promise(setImmediate)` at least once, maybe a couple of times for good measure, to force GC to happen, and allow any Finalizers to run
  * we're not using Finalizers yet, but if we were, that'd be the idiom to let them run
  * the `node` process must be started with `--expose-gc` to make `gc()` available, so add that to the package.json `ava.nodeArguments` (?? check the ava docs, I might have spelled that wrong)
  * also create a `WeakRef` to `thing1`, and assert that it's empty after the force-GC-idiom, so the test will fail if we failed to provoke enough GC to exercise the thing we're testing
* do one of the things that reanimates the vobj (and creates a new Representative): either send in a vref, or do a `.get` on a WeakStore where the value is a virtual object
  * but do *not* actually read any properties yet
* check the storage operations to make sure that no syscalls (which must be determistic) were/would-have-been made as a result of the reanimation (which happens nondeterministically)

I think our plan is to make WeakStore be just a WeakMap with the `init`-style API, rather than being a reduced-RAM serialize-the-values thing (and one of @erights 's new collections will handle the reduced-RAM use case). So maybe using a WeakStore `.get` to unserialize a virtual-object *value* is not useful: in a just-WeakMap-plus-`init` world, the *values* are held strongly, so our safety-check `WeakRef` check would fail.

Reply:
hm, maybe this sort of test belongs in the `virtualized weak collection operations` test function instead

Reply:
I think the easiest thing will be to augment the fake virtual object manager with the ability to log all the storage syscalls, the augment the 'virtual object operations' test with asserts on the log after each step.

Reply:
In general I'm uncomfortable having tests to verify that a piece of code that doesn't do something actually doesn't do it, since the number of possible things some code might not do is basically infinite, but in this case the interaction between the virtual object manager and secondary storage is complicated, and just verifying that what it actually does proceeds in lock step with our mental model of what it does seems both reasonably justified and fairly easy to implement.

Reply:
yeah, I think such a test also gives us a fighting chance of discovering any bugs that might get introduced via later refactoring.. we can't exhaustively test the way the code reacts to arbitrary input, but we can at least think of a couple of likely failure modes and include a check for them